Source/Core/DolphinWX/Input/InputConfigDiag.h

```diff
@@ -105,13 +105,13 @@ class ControlDialog : public wxDialog
 {
 public:
   ControlDialog(InputConfigDialog* const parent, InputConfig& config,
-                ControllerInterface::ControlReference* const ref);
+                ControlReference* const ref);
 
   bool Validate() override;
 
   int GetRangeSliderValue() const;
 
-  ControllerInterface::ControlReference* const control_reference;
+  ControlReference* const control_reference;
```

Comment:
Three things:
- Does this (and the `InputConfig` below) have to be public or can you just provide an accessor function for it?
- Naming convention would be to start with an `m_` prefix for member variables.
- Could you look into the possibility of making that a `std::unique_ptr` for example to get rid of the `new`/`delete` all over the place?